Light team Valid Chess moves

Mrsatatima opened the pull request:
All moves valid moves for a chess a solved only

drak00 committed:
create greetings.yml

drak00 committed:
Fix: update moves for all pieces

Mrsatatima committed:
Create Valid chess move

taking all possible moves and checking if any of the move will expose
the king... moves are then removed and updated to if the chess rules

Mrsatatima committed:
Start Pawn promotion algorithm

Bilaerl commented on engine.py:
```diff
@@ -185,6 +188,13 @@ def make_move(self, move):
 		self.board[move.end_row][move.end_col] = move.piece_moved
 		self.move_log.append(move) # log move
 		self.light_to_move = not self.light_to_move # next player to move
```
this should be the last line in this method

Bilaerl commented on engine.py:
```diff
@@ -193,20 +203,80 @@ def undo_move(self, look_ahead_mode = False):
+			it returns the updated possible valid moves and current_turn
+
+			it calls the get_possible_moves function and incheck function
+		"""
```
Use the same format Adebayo used to document method input variable(s) and return variable(s)

Bilaerl commented on engine.py:
```diff
@@ -193,20 +203,80 @@ def undo_move(self, look_ahead_mode = False):
+		"""
+		moves, turn = self.get_possible_moves() # passing in two varibles moves and turns because it returns 2
+		for i in range(len(moves)-1,-1,-1): # iterating the index of the moves from the last to 1st
+											# because we will be removing moves which will skip moves if iterated from 1st to last
```
if a comment is too long, break it into lines, and put it above the line it is explaining

Bilaerl commented on engine.py:
```diff
@@ -193,20 +203,80 @@ def undo_move(self, look_ahead_mode = False):
+			Returns a boolean
+			uses the square_under_attack fuction to test the moves that will lead to king being attacked
+			checks for both light and dark pieces
+		"""
```
documentation format inconsistency

Bilaerl commented on engine.py:
```diff
@@ -193,20 +203,80 @@ def undo_move(self, look_ahead_mode = False):
+			c -----> column of the piece position (int)
+			returns a boolean
+		"""
+		self.light_to_move = not self.light_to_move # swaping turns
```
make this comment more elaborate, it should say why you have to switch turns

Bilaerl commented on engine.py:
```diff
@@ -193,20 +203,80 @@ def undo_move(self, look_ahead_mode = False):
+		"""
+		self.light_to_move = not self.light_to_move # swaping turns
+		oppostion_moves, turn= self.get_possible_moves() # passing in two varibles moves and turns because it returns 2
+		self.light_to_move = not self.light_to_move # swaping turns
```
make this one more elaborate too

Bilaerl commented on engine.py:
```diff
@@ -193,20 +203,80 @@ def undo_move(self, look_ahead_mode = False):
+
+
+
+	def incheck(self):
```
use snake_case for method names for consistency with other methods. I.e. in_check not incheck

Bilaerl reviewed (changes requested):
Nice algorithm
I don't see any changes in main.py even though git is saying it has been change. In case I missed it, please draw my attention to it

Bilaerl commented on engine.py:
```diff
@@ -193,20 +203,80 @@ def undo_move(self, look_ahead_mode = False):
+
 			self.board[last_move.start_row][last_move.start_col] = last_move.piece_moved
 			self.board[last_move.end_row][last_move.end_col] = last_move.piece_captured
 			self.light_to_move = not self.light_to_move
```
This should also be the last line of this method

Bilaerl commented:
Please make required changes ASAP, as the person working on castling relies on methods you created (square_under_attack, in_check, e.t.c) to do their work

Mrsatatima committed:
Update Checkmate and Add Pawn promotion

Add Checkmate and Stalemate to display on screen of game board when
there are not possible moves

Add pawn promotion in queen when pawn reaches row 7 for light and row 0
for dark... room for upgrade to prompt user choice
